### ISSUE TITLE
[pickers] Accessibility issue | Date-Time Range Picker does not support text resize up to 200% as per WCAG 2.1 SC 1.4.4

### ISSUE THREAD

**Issue by swapnilchafale** [mui/mui-x#20418]
### Steps to reproduce

Steps to Reproduce:

1. Navigate to: https://mui.com/x/react-date-pickers/date-time-range-picker/calendar
2. Open the Date-Time Range Picker example.
3. Increase font size by 200% using:
Browser settings (Chrome → Settings → Appearance → Font Size → Very Large) OR
OS Accessibility Settings (e.g., Windows → Display → Scale text size) OR
any browser extension to text resize up to 200%,  example - (A+ FontSize Changer).
4. Observe the calendar and time selection UI.

### Current behavior

When testing the Date-Time Range Picker component on MUI Date-Time Range Picker, the text does not properly resize when font size is increased by 200% using browser or OS accessibility settings. This violates WCAG 2.1 Success Criterion 1.4.4 (Resize Text), which requires that text can be resized up to 200% without loss of content or functionality.

1. Text overlaps and gets truncated in the time picker and calendar view.
2. Labels such as AM/PM, dates, and time values are partially hidden.
3. Interactive elements become misaligned and difficult to use.

<img width="1712" height="1944" alt="Image" src="https://github.com/user-attachments/assets/0d5154ac-89a9-41dc-8d94-221306d62bc5" />

<img width="1712" height="1944" alt="Image" src="https://github.com/user-attachments/assets/f78124b8-2f43-49d1-98bb-3c025f3e7854" />

### Expected behavior

1. All text should resize up to 200% without clipping, overlapping, or loss of functionality.
2. Layout should adjust dynamically to accommodate larger text.

### Context

- Users with low vision who rely on text resizing cannot effectively use the date-time picker.
- This creates a barrier for accessibility compliance.

WCAG Reference: 1.4.4 Resize Text: Text can be resized up to 200% without assistive technology and without loss of content or functionality.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: Pickers, Accessibility, WCAG, Date-Time Pickers

**Comment by michelengelen** [comment]
This is a valid request, so I'll add this to the board for the team to have a look and come up with a solution.
